KA-89

bhirmbani opened the pull request:
WIP
- [x] init upload endpoint works
- [x] make bucket name: username/email, Key: ${prexif_by_upload_type}.${extension}
- [x] conditional on uploadImg method to accomodate upload img based on purpose (id/user photo)
- [x] make sure uploadImg returning s3 link to store at courierProposal table

bhirmbani committed:
adding test image

bhirmbani committed:
adding aws s3 service file

bhirmbani committed:
adding upload img route

bhirmbani committed:
remove unused dependencies

bhirmbani committed:
initializing uploadImg method

bhirmbani committed:
merge from develop github

bhirmbani committed:
adding check token endpoint

bhirmbani committed:
delete unused migration file

bhirmbani committed:
remove foreignKey config in user and courier proposal association so include can work

bhirmbani committed:
updating courier-proposal migration file

bhirmbani committed:
adding working get proposal method

bhirmbani committed:
clean

bhirmbani committed:
fix wrong key related to CourierProposal model

bhirmbani committed:
merge from KA-87

bhirmbani committed:
refactor s3 service

bhirmbani committed:
add upload img route

bhirmbani committed:
update bodyparser setup to accomodate base64 upload img

bhirmbani committed:
moving method for proposal handle to service

bhirmbani committed:
moving to service, upload img to s3 using base64, update with link in courierproposal model

bhirmbani committed:
update msg

bhirmbani commented:
ini yg paling update kayanya @andy-shi88

bhirmbani committed:
return updated data in updatesenderproposal endpoint

bhirmbani committed:
change proposal middleware to sysadmin

bhirmbani committed:
delete password if get user, change get user route middleware to sysadmin

andy-shi88 commented:
wah, baru baca, paling update gimna maksudny? @bhirmbani

andy-shi88 commented on controllers/UserController.js:
```diff
@@ -61,8 +65,12 @@ export default class UserController {
     try {
-      uniqueEmail = await this.service.findOne({ email });
-      uniqueUsername = await this.service.findOne({ username });
+      uniqueEmail = await this.service.findOne({
```
ini ada 2 kali query, bisa disatukan dengan operator OR

bhirmbani replied:
ini sengaja aku satukan, Dy. kalo pake operator OR, untuk menentukan message nya ga bisa nentuin yg sudah ada di db apakah email/username.

andy-shi88 commented on controllers/UserController.js:
```diff
@@ -506,4 +569,206 @@ export default class UserController {
+            return data;
+          });
+          if (type === 'ID') {
+            await this.service.proposeModel.update(
```
ini coba pas nanti resolve conflict dari develop di pindahkan ke proposal controller.

andy-shi88 reviewed (changes requested):
ini coba kamu ambil dari develop dulu, resolve conflict, yang ada `this.service.proposeModel` pindahin ke proposal controller logicnya. untuk yang forgot password udah kurecover, kalau mau memudahkan resolve conflict, bisa di merge dulu

bhirmbani commented:
oke, yang branch forgot password sya merge dlu ya

bhirmbani committed:
comment

bhirmbani committed:
merge from remote develop branch

bhirmbani committed:
move courier proposal service

bhirmbani committed:
add options in update method in base service

bhirmbani committed:
add sysadmin middleware in get user, remove get proposal

bhirmbani committed:
renaming middleware helper from admin to siteadmin

bhirmbani committed:
deleting unused admin route

bhirmbani committed:
moving proposal method in user controller to proposal controller

bhirmbani committed:
renaming

bhirmbani committed:
renaming proposal route

bhirmbani committed:
updating proposal test

bhirmbani committed:
update test

bhirmbani committed:
remove unused var

bhirmbani committed:
delete unused Mail helper

bhirmbani committed:
change babel preset 2015 to env, include migration folder to babel build

preparint to test deploy on heroku

bhirmbani committed:
updating npm script, procfile for heroku deploy

bhirmbani commented:
@andy-shi88 Minta review nya ya.

bhirmbani committed:
delete yarn.lock

bhirmbani committed:
update package #3

bhirmbani committed:
Merge branch 'develop' of https://github.com/gamantha/kurir-app-server into KA-89

bhirmbani committed:
updating package and procfile

bhirmbani committed:
adding npm run build on package

bhirmbani committed:
updating .babelrc file

this update make sure importing json file and support async await function

bhirmbani committed:
updating mail route address

for consistency

bhirmbani committed:
tweak index file on /models

so babel can import the json file on build

bhirmbani committed:
adding required babel plugin

this plugin make sure json file imported and async await supported

bhirmbani committed:
change config from require to import

so babel build can interpret it

bhirmbani committed:
adding dist folder to git, update package

bhirmbani committed:
adding dist folder

bhirmbani committed:
updating baseUrl

for sending verification link to email

bhirmbani committed:
exclude test and seeders file in dist

bhirmbani committed:
updating dist file

with heroku base url, remove test and seeders folder

bhirmbani committed:
run heroku using nodemon

bhirmbani committed:
won't send another welcome email, updating msg

bhirmbani committed:
updating dist

so user won't receive another welcome email if already validated, also update msg

bhirmbani committed:
updating check email controller

bhirmbani committed:
update dist again

for check email controller update

bhirmbani committed:
adding npm run build on clean

bhirmbani committed:
update check email route

bhirmbani committed:
fixing bug in update proposal

bhirmbani committed:
updating dist

bhirmbani committed:
Merge branch 'develop' of https://github.com/gamantha/kurir-app-server into heroku

bhirmbani committed:
update change pass controller in dist

bhirmbani committed:
init wallaby config

bhirmbani committed:
message on test, updating change password

bhirmbani committed:
updating models

to accomodate user has one proposal

bhirmbani committed:
updating route

singular for consistency

bhirmbani committed:
updating vscode

init wallabyjs

bhirmbani committed:
updating test db config

bhirmbani committed:
updating proposal route

bhirmbani committed:
updating proposal controller and service

bhirmbani committed:
updating change password with params

fixing bugs in params forgotpassword

bhirmbani committed:
adding new constant in controller test

bhirmbani committed:
clean

bhirmbani committed:
update babelrc and change to env

bhirmbani committed:
fix conflict after merge KA-89 to develop

bhirmbani committed:
adding babel env and other plugin, fix package json

bhirmbani committed:
merge from KA-89 to develop

bhirmbani committed:
merge develop into heroku

bhirmbani committed:
clean

bhirmbani committed:
updating dist

bhirmbani committed:
delete /dist

bhirmbani committed:
updating change pass routes

bhirmbani committed:
updating change pass route in dist

bhirmbani committed:
updating check code pass route

andy-shi88 commented:
@bhirmbani di merge aja kalau udah clear 👍

bhirmbani committed:
rename check & forgot pass method

bhirmbani committed:
delete unused mail route

bhirmbani committed:
rename send email validation link route

bhirmbani committed:
merge to KA-89 from heroku local

bhirmbani committed:
delete dist

bhirmbani committed:
update login test, comment forgot pass test
